App/Syncthing/syncthing_sync_check.sh

```diff
@@ -0,0 +1,284 @@
+    local response=$(curl -s -X GET -H "X-API-Key: $API_KEY" "$API_ENDPOINT/system/discovery")
+    if [ $? -eq 0 ]; then
+        log_message "Initiated device rediscovery"
+    else
+        log_message "Error: Failed to initiate device rediscovery"
+    fi
+}
+
+stop_network() {
+    log_message "Stopping network interface..."
+    ifconfig lo down
+}
+
+get_folders() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/config/folders" | jq -r '.[] | "\(.id)|\(.label)"'
+}
+
+get_devices() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/system/connections" | jq -r '.connections | keys[]'
+}
+
+are_devices_online() {
+    local connected_devices=$(curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/system/connections" | jq -r '.connections | to_entries[] | select(.value.connected == true) | .key')
+    if [ -n "$connected_devices" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+get_folder_status() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/db/status?folder=$1"
+}
+
+get_device_name() {
+    local device_id="$1"
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/config/devices" | jq -r ".[] | select(.deviceID == \"$device_id\") | .name"
+}
+
+calculate_folder_completion() {
+    local folder_id="$1"
+    local folder_status=$(get_folder_status "$folder_id")
+    local global_bytes=$(echo "$folder_status" | jq '.globalBytes // 0')
+    local need_bytes=$(echo "$folder_status" | jq '.needBytes // 0')
+
+    if [ "$global_bytes" -eq 0 ]; then
+        echo "100"
+    else
+        local completed_bytes=$((global_bytes - need_bytes))
+        local percentage=$((completed_bytes * 100 / global_bytes))
+        echo $percentage
+    fi
+}
+
+get_completion() {
+    curl -s -H "X-API-Key: $API_KEY" "$API_ENDPOINT/db/completion?device=$1&folder=$2"
+}
+
+calculate_total_completion() {
+    local device="$1"
+    local folder_id="$2"
+    local completion=$(get_completion "$device" "$folder_id")
+
+    local global_bytes=$(echo "$completion" | jq '.globalBytes // 0')
+    local need_bytes=$(echo "$completion" | jq '.needBytes // 0')
+    local need_deletes=$(echo "$completion" | jq '.needDeletes // 0')
+    local need_items=$(echo "$completion" | jq '.needItems // 0')
+
+    if [ "$need_deletes" -gt 0 ] || [ "$need_items" -gt 0 ] || [ "$need_bytes" -gt 0 ]; then
+        local percentage=$((100 * (global_bytes - need_bytes) / global_bytes))
+        echo $percentage
+    else
+        echo "100"
+    fi
+}
+
+monitor_sync_status() {
+    local mode="$1"
+    local folders=$(get_folders)
+    local devices=$(get_devices)
+
+    log_message "Monitoring sync status in $mode mode"
+    display -t "Syncthing Check:
+
+Press START to cancel"
+
+    if [ "$mode" = "shutdown" ]; then
+        force_rescan
+        sleep 1
+    elif [ "$mode" = "startup" ]; then
+        force_rediscovery
+        # THIS MIGHT NOT BE NEEDED, WE MIGHT BE ABLE TO JUST SLEEP 1
+        # sleep 1
+        sleep 5
```

Comment:
If it's not needed, it seems better to sleep 1 than 5.

Reply:
Definitely, I wanted to do another round of testing with this sleep changed and see if it's enough.

This originally was set to 5 seconds because I was calling the force_rediscovery endpoint correctly. I'll do more testing and report back (I definitely wont merge this without that test)

Reply:
I did some testing and 2 seconds was the magic number. 1 second worked most of the time but 2 seconds worked all the time.
https://github.com/spruceUI/spruceOS/pull/110/commits/807be37a3ebd5ae13b0907f761e95c12fe247497